ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/SandboxOnXRunner.scala

```diff
@@ -0,0 +1,324 @@
+  }
+
+  private def validateCombinedParticipantMode(config: Config[BridgeConfig]) =
+    config.participants.toList match {
+      case participantConfig :: Nil if participantConfig.mode == ParticipantRunMode.Combined =>
+        Resource.successful(participantConfig)
+      case _ =>
+        Resource.failed {
+          val loggingMessage = "Sandbox-on-X can only be run in a single COMBINED participant mode."
+          newLoggingContext(logger.info(loggingMessage)(_))
+          new IllegalArgumentException(loggingMessage)
+        }
+    }
+
+  private def buildLedger(
+      sharedEngine: Engine
+  )(implicit
+      config: Config[BridgeConfig],
+      participantConfig: ParticipantConfig,
+      materializer: Materializer,
+      actorSystem: ActorSystem,
+  ) = {
+    implicit val apiServerConfig: ApiServerConfig =
+      BridgeConfigProvider.apiServerConfig(participantConfig, config)
+
+    newLoggingContextWith("participantId" -> participantConfig.participantId) {
+      implicit loggingContext =>
+        for {
+          metrics <- buildMetrics
+          translationCache = LfValueTranslationCache.Cache.newInstrumentedInstance(
+            eventConfiguration = config.lfValueTranslationEventCache,
+            contractConfiguration = config.lfValueTranslationContractCache,
+            metrics = metrics,
+          )
+
+          (stateUpdatesFeedSink, stateUpdatesSource) <- AkkaSubmissionsBridge()
+
+          servicesExecutionContext <- buildServicesExecutionContext(metrics)
+
+          readServiceWithSubscriber = new BridgeReadService(
+            ledgerId = config.ledgerId,
+            maxDedupSeconds = config.extra.maxDedupSeconds,
+            stateUpdatesSource,
+          )
+
+          indexerHealthChecks <- buildIndexerServer(
+            metrics,
+            servicesExecutionContext,
+            new TimedReadService(readServiceWithSubscriber, metrics),
+            translationCache,
+          )
+
+          indexService <- standaloneIndexService(
+            sharedEngine,
+            config,
+            apiServerConfig,
+            metrics,
+            translationCache,
+            servicesExecutionContext,
+          )
+
+          writeService <- buildWriteService(stateUpdatesFeedSink, servicesExecutionContext)
+
+          _ <- buildStandaloneApiServer(
+            sharedEngine,
+            indexService,
+            metrics,
+            servicesExecutionContext,
+            new TimedWriteService(writeService, metrics),
+            indexerHealthChecks,
+          )
+        } yield ()
+    }
+  }
+
+  private def standaloneIndexService(
+      sharedEngine: Engine,
+      config: Config[BridgeConfig],
+      apiServerConfig: ApiServerConfig,
+      metrics: Metrics,
+      translationCache: LfValueTranslationCache.Cache,
+      servicesExecutionContext: ExecutionContextExecutorService,
+  )(implicit loggingContext: LoggingContext, materializer: Materializer) =
+    for {
+      dbSupport <- DbSupport
```

Comment:
FYI: this will be needed by PersistentUserManagementStore (currently inmem with the comment)
no need to do anything, @pbatko-da can also refactor later